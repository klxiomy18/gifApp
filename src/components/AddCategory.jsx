import PropTypes from "prop-types";

import React, { useState } from 'react';

 export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('One punch');

    const onInputChange = (event) => {

        setInputValue(event.target.value)

    }

    const onSubmit = (event) => {
        event.preventDefault();

        if(inputValue.trim().length <= 1) return

        // setCategories((categories) => [inputValue, ...categories]);
        onNewCategory(inputValue.trim())
        setInputValue('')
    }

  return (
   
    <form aria-label="form" action="" onSubmit={onSubmit}>
        <input 
        type='text'
        placeholder='Buscar gifs'
        value={inputValue}
        onChange={ onInputChange}
    />
    </form>
    
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired
}