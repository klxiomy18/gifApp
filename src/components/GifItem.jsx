import PropTypes from "prop-types";

import React from 'react'

export const GifItem = ({url, title}) => {
  return (
    <div className='card'>
        <img src={url} alt={title} />
        <p>{title}</p>
    </div>
  )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}
