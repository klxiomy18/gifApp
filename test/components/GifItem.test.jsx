import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"


describe('Pruebas en GifItem', () => {
    const title = 'Saitama'; 
    const url = 'https://saitama'
    test('Hacer match con el sanpshot', () => {
      const {container} = render(<GifItem url={url} title={title}/>);
      expect(container).toMatchSnapshot()
    })
    
    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {
        
      render(<GifItem url={url} title={title}/>);
      // screen.debug();
      // expect(screen.getByRole('img').src).toContain(url);
      // expect(screen.getByRole('img').alt).toContain(title);
      const {alt, src} = screen.getByRole('img');

      expect(src).toContain(url);
      expect(alt).toContain(title);
    });

    test('Debe de mostrr el titulo en el componente', () => {

      render(<GifItem url={url} title={title}/>);
      
      expect(screen.getByText(title)).toBeTruthy();
    })
    
})

