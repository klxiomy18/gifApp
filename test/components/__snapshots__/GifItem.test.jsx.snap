// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Hacer match con el sanpshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://saitama"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
